packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/IrUtils.kt

```diff
@@ -531,10 +531,15 @@ fun getBacklinksTargetPropertyType(declaration: IrProperty): IrType? {
     }
 }
 
-fun getLinkingObjectPropertyName(backingField: IrField): Name {
+fun getLinkingObjectPropertyName(property: IrProperty): String {
+    if (property.hasAnnotation(PERSISTED_NAME_ANNOTATION)) {
+        val annotation: IrConstImpl<String> = property.getAnnotation(PERSISTED_NAME_ANNOTATION).getValueArgument(0) as IrConstImpl<String>
```

Comment:
I am not sure if this is the nicest way to do this in Kotlin, is there a better syntax which would do a check whether annotation exists and then do a cast and return value?

Reply:
outdated: will most likely change this back